src/atlas/projection/detail/StretchRegular.h

```diff
@@ -0,0 +1,78 @@
+/**
+* (C) Crown copyright 2021, Met Office
+*
+* This software is licensed under the terms of the Apache Licence Version 2.0
+* which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+*/
+
+#pragma once
+
+#include "atlas/domain.h"
+#include "atlas/projection/detail/ProjectionImpl.h"
+#include "atlas/util/NormaliseLongitude.h"
+
+namespace atlas {
+namespace projection {
+namespace detail {
+
+template <typename Rotation>
+class StretchLAM final : public ProjectionImpl {
+public:
+    using Spec = ProjectionImpl::Spec;
+
+    ///< constructor uses parametrisation and point to stretch
+    StretchLAM(const eckit::Parametrisation& );
+    ///< projection name
+    static std::string static_type() { return Rotation::typePrefix() + "stretch"; }
+    std::string type() const override { return static_type(); }
+
+    ///< projection and inverse projection
+    void xy2lonlat( double crd[] ) const override;
+    void lonlat2xy( double crd[] ) const override;
+
+    ///< specification for stretching
+    Spec spec() const override;
+
+    std::string units() const override { return "meters"; }
```

Comment:
The units should be "degrees" if I follow the code correctly.

Reply:
done